autogen/cluster.tf

```diff
@@ -219,14 +219,31 @@ resource "google_container_cluster" "primary" {
 /******************************************
   Create Container Cluster node pools
  *****************************************/
+resource "random_id" "name" {
+  # if any node_pool definition has a create_before_destroy key, then create random_id names
+  count = length(compact([for node_pool in var.node_pools : lookup(node_pool, "create_before_destroy", "")])) > 0 ? length(var.node_pools) : 0
+
+  byte_length = 2
+  prefix      = format("%s-", lookup(var.node_pools[count.index], "name"))
+
+  keepers = {
```

Comment:
How are we tracking which fields force a recreate and which can be updated in-place?

Reply:
I am not exactly sure how to track which resources should be added to keepers other than looking at the node pool schema and seeing which ones have "ForceNew" set. What do you think about adding any resources specified with `ForceNew: true` in the [schemaNodeConfig](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/node_config.go#L22) variable to keepers? 

Then adding a comment above keepers to signify which keepers how to determine which resources to add.

Reply:
Following the `ForceNew: true` method, one way to signify to track the keepers is through a local variable like `resources_which_force_node_pool_recreation` and iterating through the list to lookup those resources in the node_pool. 

```hcl
locals {
  resources_which_force_node_pool_recreation = [
    "disk_size_gb",
    "disk_type",
    "accelerator_count",
    "accelerator_type",
    "local_ssd_count",
    "machine_type",
    "preemptible",
    "service_account",
  ]
}

resource "random_id" "name" {
  count       = length(var.node_pools)
  byte_length = 2
  prefix      = format("%s-", lookup(var.node_pools[count.index], "name"))
  keepers     = zipmap(
      local.resources_which_force_node_pool_recreation,
      [ for keeper in local.resources_which_force_node_pool_recreation : lookup(var.node_pools[count.index], keeper, "") ]
  )
}
```

The problem that I am running into is the list of map resources such as `labels`, `metadata`, `oauth_scopes`, and `tags` which do not conform to the same pattern as these resources are a result from merging the `all` tag with the node_pool config. 

What do you think about retaining a small list of keepers with a local such as `resources_which_force_node_pool_recreation` ?

Reply:
The latest commit tracks all resources including `labels`, `metadata`, oauth_scopes`, and `tags` by converting each map or list into a string and creating a map for keepers.

Reply:
Perfect, thanks. Could we add a comment that this keepers list should be based on the `ForceNew` params?

Reply:
added.